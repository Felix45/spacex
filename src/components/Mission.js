import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';

const Mission = (props) => {
  const { mission } = props;
  const { mission_id, mission_name, description } = mission;

  return (
    <tr>
      <td className="fw-bold">{ mission_name }</td>
      <td>{ description }</td>
      <td className="mx-1">
        <Badge bg="secondary">Not a member</Badge>
      </td>
      <td className="p-2">
        <Button id={mission_id} variant="outline-dark" className="p-0" size="sm">Join Mission</Button>
      </td>
    </tr>
  );
};

Mission.propTypes = {
  mission: PropTypes.instanceOf(Object).isRequired,
};

export default Mission;
