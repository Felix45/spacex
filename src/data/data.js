const ships = [
  {
    id: 1,
    rocket_name: 'Falcon1',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://i.imgur.com/DaCfMsj.jpg',
  },
  {
    id: 2,
    rocket_name: 'Falcon9',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
  },
  {
    id: 3,
    rocket_name: 'Starship',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg',
  },
];

export default ships;
