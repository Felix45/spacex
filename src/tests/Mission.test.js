import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux';
import { missionData } from '../data/mock';
import Mission from '../components/Mission';

test('Mission component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store({})}>
      <Mission mission={missionData[1]} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
