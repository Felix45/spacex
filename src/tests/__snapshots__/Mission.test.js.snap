// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly 1`] = `
<tr>
  <td
    className="fw-bold"
  >
    Telstar
  </td>
  <td>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
  </td>
  <td
    className="align-middle"
  >
    <span
      className="text-uppercase badge bg-info"
    >
       Active Member 
    </span>
  </td>
  <td
    className="align-middle"
  >
    <button
      className="w-100 text-uppercase btn btn-outline-danger"
      disabled={false}
      id={2}
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
